src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.spec.ts

```diff
@@ -1,33 +1,505 @@
+    courseService = TestBed.inject(CourseService);
+    feedbackSessionsService = TestBed.inject(FeedbackSessionsService);
+    feedbackQuestionsService = TestBed.inject(FeedbackQuestionsService);
+    studentService = TestBed.inject(StudentService);
+    instructorService = TestBed.inject(InstructorService);
+    navigationService = TestBed.inject(NavigationService);
+    ngbModal = TestBed.inject(NgbModal);
     component = fixture.componentInstance;
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should snap with default fields', () => {
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap with feedback session questions', () => {
+    component.questionEditFormModels = [testQuestionEditFormModel1, testQuestionEditFormModel2];
+    component.isLoadingFeedbackSession = false;
+    component.isLoadingFeedbackQuestions = false;
+    component.courseId = testCourse1.courseId;
+    component.courseName = testCourse1.courseName;
+    component.feedbackSessionName = testFeedbackSession.feedbackSessionName;
+    component.sessionEditFormModel = sessionEditFormModel;
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap with new question added', () => {
+    component.isAddingQuestionPanelExpanded = true;
+    component.isLoadingFeedbackSession = false;
+    component.isLoadingFeedbackQuestions = false;
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap when feedback session failed to load', () => {
+    component.hasLoadingFeedbackSessionFailed = true;
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap when feedback question failed to load', () => {
+    component.hasLoadingFeedbackQuestionsFailed = true;
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap when feedback session is loading', () => {
+    component.isLoadingFeedbackSession = true;
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap when feedback questions are loading', () => {
+    component.isLoadingFeedbackQuestions = true;
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should load correct feedback session for a given API output', () => {
+    spyOn(courseService, 'getCourseAsInstructor').and.returnValue(of(testCourse1));
+    spyOn(feedbackSessionsService, 'getFeedbackSession').and.returnValue(of(testFeedbackSession));
+
+    component.loadFeedbackSession();
+
+    expect(component.courseName).toBe('Test Course 1');
+    expect(component.sessionEditFormModel.courseId).toBe('testId1');
+    expect(component.sessionEditFormModel.courseName).toBe('Test Course 1');
+    expect(component.sessionEditFormModel.feedbackSessionName).toBe('Test Session');
+  });
+
+  it('should load correct feedback session questions', () => {
```

Comment:
What about testing the error cases for all these endpoints?

Reply:
Added error tests for endpoints